lib/Dialect/Handshake/HandshakeOps.cpp

```diff
@@ -364,6 +412,26 @@ static ParseResult verifyFuncOp(handshake::FuncOp op) {
   if (failed(verifyPortNameAttr("resNames", op.getNumResults())))
     return failure();
 
+  // Verify that block arguments have single uses.
+  for (auto barg : enumerate(entryBlock.getArguments())) {
+    if (!hasSingleUse(barg.value()))
+      return op.emitOpError()
+             << "argument " << barg.index() << " has multiple uses.";
+  }
+
+  // Verify that any non-handshake operation result has a single use. Handshake
+  // operations themselves validate this through the HasSingleUseResults trait.
+  for (auto &subOp : op.getOps()) {
+    if (subOp.getDialect()->getNamespace() ==
```

Comment:
I think you can say `isa<HandshakeDialect>(subOp.getDialect())`

Reply:
Nice, will do!